pkg/lib/cockpit-components-modifications.jsx

```diff
@@ -37,85 +37,77 @@
  * Enables showing shell and ansible script. Shell one is mandatory and ansible one can be omitted.
  *
  */
-class ModificationsExportDialog extends React.Component {
-    constructor(props) {
-        super(props);
+const ModificationsExportDialog = ({ show, onClose, shell, ansible }) => {
+    const [active_tab, setActiveTab] = React.useState("shell");
+    const [copied, setCopied] = React.useState(false);
+    const [timeoutId, setTimeoutId] = React.useState(null);
 
-        this.timeoutId = null;
-        this.state = {
-            active_tab: "shell",
-            copied: false
-        };
-
-        this.handleSelect = this.handleSelect.bind(this);
-        this.copyToClipboard = this.copyToClipboard.bind(this);
-    }
-
-    handleSelect(event, active_tab) {
-        this.setState({ active_tab, copied: false });
-        if (this.timeoutId !== null) {
-            clearTimeout(this.timeoutId);
-            this.timeoutId = null;
+    const handleSelect = (_event, active_tab) => {
+        setCopied(false);
+        setActiveTab(active_tab);
+        if (timeoutId !== null) {
+            clearTimeout(timeoutId);
+            setTimeoutId(null);
         }
-    }
+    };
 
-    copyToClipboard() {
+    const copyToClipboard = () => {
         try {
-            navigator.clipboard.writeText(this.props[this.state.active_tab].trim())
+            navigator.clipboard.writeText((active_tab === "ansible" ? ansible : shell).trim())
                     .then(() => {
-                        this.setState({ copied: true });
-                        this.timeoutId = setTimeout(() => {
-                            this.setState({ copied: false });
-                        }, 3000);
+                        setCopied(true);
+                        setTimeoutId(setTimeout(() => {
+                            setCopied(false);
+                            setTimeoutId(null);
+                        }, 3000));
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19370-20230922-100423-663bbd9c-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)